discord-bot/src/utils/config.ts

```diff
@@ -0,0 +1,27 @@
+import * as dotenv from 'dotenv'
+
+dotenv.config()
+
+const loadEnv = (key: string): string => {
+  if (!process.env[key]) throw new Error(`${key} is not defined`)
+  return process.env[key] || ''
+}
+
+export const config = {
+  TOKEN: loadEnv('TOKEN'),
+  GUILD_ID: loadEnv('GUILD_ID'),
+  CLIENT_ID: loadEnv('CLIENT_ID'),
+  CONTRACT_ADDRESS: loadEnv('CONTRACT_ADDRESS'),
+  PRIVATE_KEY: loadEnv('PRIVATE_KEY'),
```

Comment:
I'm not very experienced with TS, is there any potential security issue with exporting `config` here rather than fetching the values directly from `.env` file where necessary? 
The logic seems a bit strange to me.

Reply:
The issue is in TS if you simply use `process.env.something` this will be a `string | undefined` and most of the functions you are passing the env to won't allow undefined, so you will need to add sanity check everywhere, furthermore you will know if you are missing an env only when this variable is in scope.

Loading them all at once allow us to verify that all these variables exist when the bot launch, and in term of security, I don't see a risk here, the only input is a string, the address the user passes to the command, this is a string that is used to call the contract. it's not like SQL with potential SQL injection, if you try to pass any special character the call to the contract will simply fail and the user will get an error msg.

The main security thread I can see happening is us leaking the private key the bot has access or the container or instance the bot is run on getting access, and then that private key compromise. This is why I made a Minter role in the Faucet contract, so that the private key can't withdraw all the tokens hold by the contract, and can only be used to execute the request token function.

So if we see the balance of the contract going down faster than the amount of tx log on Slack, we can revoke the minter role and rotate keys

Reply:
That makes sense, thanks for the clarification!